crates/revm/revm-inspectors/src/tracing/builder/parity.rs

```diff
@@ -154,14 +160,28 @@ impl ParityTraceBuilder {
         DB: DatabaseRef,
     {
         let ResultAndState { result, state } = res;
+
+        let mut depth_first_call_addresses = CallTraceNodeWalkerDF::new(&self.nodes)
+            .map(|node| node.trace.caller)
+            .collect::<Vec<_>>()
+            .into_iter();
+
         let mut trace_res = self.into_trace_results(result, trace_types);
+
+        // check the state diff case
         if let Some(ref mut state_diff) = trace_res.state_diff {
             populate_account_balance_nonce_diffs(
                 state_diff,
                 &db,
                 state.into_iter().map(|(addr, acc)| (addr, acc.info)),
             )?;
         }
+
+        // check the vm trace case
+        if let Some(ref mut vm_trace) = trace_res.vm_trace {
+            populate_vm_trace_bytecodes(&db, &mut depth_first_call_addresses, vm_trace)?;
+        }
```

Comment:
we should be able to optimize this here

if no vmtrace then no walker: empty array
else use walker the array